kafkaesk/ext/logging/handler.py

```diff
@@ -0,0 +1,134 @@
+from .formatter import PydanticLogModel
+from .record import PydanticLogRecord
+from pydantic import BaseModel
+from typing import IO
+from typing import Optional
+from typing import Tuple
+
+import asyncio
+import kafkaesk
+import logging
+import sys
+
+
+class InvalidLogFormat(Exception):
+    ...
+
+
+class PydanticStreamHandler(logging.StreamHandler):
+    def __init__(self, stream: Optional[IO[str]] = None):
+        super().__init__(stream=stream)
+
+    def format(self, record: logging.LogRecord) -> str:
+        message = super().format(record)
+
+        if isinstance(message, BaseModel):
+            message = message.json()
+
+        return message
+
+
+class KafkaeskQueue:
+    def __init__(
+        self,
+        app: kafkaesk.app.Application,
+        loop: asyncio.AbstractEventLoop = None,
+        max_queue: int = 10000,
+    ):
+        self._queue: asyncio.Queue[Tuple[str, BaseModel]] = asyncio.Queue(maxsize=max_queue)
+        self._app = app
+
+        self._app.on("finalize", self.flush)
+
+        self._loop = loop or asyncio.get_event_loop()
+        self._task: Optional[asyncio.Task] = None
+
+    def start(self) -> None:
+        if self._task is None or self._task.done():
+            self._task = self._loop.create_task(self._run())
+
+    def close(self) -> None:
+        if not self._loop.is_closed() and self._task is not None:
+            if not self._task.done() and not self._task.cancelled():
+                self._task.cancel()
+
+    async def _run(self) -> None:
+        while True:
+            try:
+                stream, message = await asyncio.wait_for(self._queue.get(), 1)
+                await self._publish(stream, message)
+
+            except asyncio.TimeoutError:
+                continue
+
+            except asyncio.CancelledError:
+                await self.flush()
+                return
+
+    async def flush(self) -> None:
+        while not self._queue.empty():
+            stream, message = await self._queue.get()
+            await self._publish(stream, message)
+
+    async def _publish(self, stream: str, message: BaseModel) -> None:
+        if self._app._intialized:
+            try:
+                await self._app.publish(stream, message)
+            except kafkaesk.exceptions.UnregisteredSchemaException:
+                self._print_to_stderr(message, "Log schema is not registered")
+            # TODO: Handle other Kafak errors that may be raised
+        else:
+            self._print_to_stderr(message, "Kafkaesk application is not initialized")
+
+    def _print_to_stderr(self, message: BaseModel, error: str) -> None:
+        sys.stderr.write(f"Error sending log to Kafak: \n{error}\nMessage: {message.json()}")
+
+    def put_nowait(self, stream: str, message: PydanticLogModel) -> None:
+        self._queue.put_nowait((stream, message))
```

Comment:
cosmetic stuff, an exception telling you that the queue is full can happen, maybe we want to catch that exception and just sent a message to the `stderr`